src/main.cpp

```diff
@@ -5238,8 +5238,13 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
             if (inv.type == MSG_TX) {
                 inv.type |= nFetchFlags;
+                pfrom->AddInventoryKnown(inv);
+                if (fBlocksOnly)
+                    LogPrint("tx", "transaction (%s) inv sent in violation of protocol peer=%d\n", inv.hash.ToString(), pfrom->id);
```

Comment:
In this case, you are being notified about network issue - some peer violated protocol (with some tx). Ping pong between tx and net has no reason here...